common/presenters/table/object-to-table-head.js

```diff
@@ -1,17 +1,61 @@
 const { pickBy } = require('lodash')
+const queryString = require('query-string')
 
 const i18n = require('../../../config/i18n')
 
-function objectToTableHead({ head } = {}) {
-  if (!head) {
-    return undefined
+function buildUrl({ key }, query = {}) {
+  return queryString.stringify({
+    ...query,
+    sortBy: key,
+    sortDirection: query.sortDirection === 'asc' ? 'desc' : 'asc',
+  })
+}
+
+function buildAriaLabel(head, query) {
+  // the label indicates the result of the sorting action triggered by the link,
+  // which is the opposite of the current state
+  return `${i18n.t('sort')} ${i18n.t(head.html).toLowerCase()} ${
+    query.sortDirection === 'asc' ? i18n.t('descending') : i18n.t('ascending')
+  }`
+}
+
+function buildWrapper(head, query) {
+  const label = i18n.t(head.html)
+  if (!head.isSortable) {
+    return label
+  }
+  const url = buildUrl(head, query)
+  const ariaLabel = buildAriaLabel(head, query)
+  return `<a aria-label="${ariaLabel}" class="sortable-table__button" role="button" href="?${url}">${label}</a>`
+}
+
+function assignSortAttributes(head, query) {
+  if (!head.isSortable) {
+    return head.attributes ? { ...head.attributes } : undefined
+  }
+  const isCurrentSortColumn = head.key === query.sortBy
+  return {
+    ...head.attributes,
+    'aria-sort': isCurrentSortColumn
+      ? query.sortDirection === 'asc'
+        ? 'ascending'
+        : 'descending'
+      : 'none',
   }
+}
 
-  return pickBy({
-    ...head,
-    html: head.html ? i18n.t(head.html) : undefined,
-    text: head.text ? i18n.t(head.text) : undefined,
-  })
+function objectToTableHead(query) {
+  return function objectToTableHead({ head } = {}) {
+    if (!head) {
+      return undefined
+    }
+    head.attributes = assignSortAttributes(head, query)
```

Comment:
This could be combined within the return to make it a little easier to read:

```javascript
    return pickBy({
      ...head,
      attributes: assignSortAttributes(head, query),
      html: head.html ? buildWrapper(head, query) : undefined,
      text: head.text ? i18n.t(head.text) : undefined,
    })
```